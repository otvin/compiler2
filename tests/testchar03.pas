program testchar03(output);
{test character passed byref}

procedure plusone(var c:char);
var i:integer;
begin
	i := ord(c);
	i := i + 1;
	c := chr(i);
end;

procedure printplusone(c:char);
var x:char;
begin
	x := c;
	writeln('before: ', x);	
	plusone(x);
	writeln('after: ', x);
end;

begin
	printplusone('0');
	printplusone('a');
	printplusone('%');
end.

	
