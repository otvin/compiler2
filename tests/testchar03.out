before: 0
after: 1
before: a
after: b
before: %
after: &
